frontend/isoko/src/features/dashboard/DashboardSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Business } from '../../types/GlobalTypes';
+import { fetchBusinessDetails, updateBusinessDetails } from './DashboardAPI';
+
+export interface DashboardState {
+   business: Business | null;
+   analytics: null;
+   status: 'idle' | 'loading' | 'failed';
+}
+
+const initialState: DashboardState = {
+   business: null,
+   analytics: null,
+   status: 'idle',
+};
+
+export const initializeBusinessDetailsAsync = createAsyncThunk(
+   'dashboard/initializeBusinessDetails',
+   async (businessId: string) => {
+      return await fetchBusinessDetails(businessId);
+   }
+);
+
+export interface UpdateParams {
+   businessId: string;
+   [key: string]: any;
```

Comment:
Maybe change the typing from any to string | Hours | Day or whatever else it can be on the right side rather than leaving it ambiguous.